lib/Infer/DataflowPruning.cpp

```diff
@@ -106,13 +133,41 @@ std::vector<ValueCache> DataflowPruningManager::generateInputSets(
   }
   InputSets.push_back(Cache);
 
-  Current = 2*Current + 1;
   for (auto &&I : Inputs) {
     if (I->K == souper::Inst::Var)
-      Cache[I] = {llvm::APInt(I->Width, Current++)};
+      Cache[I] = {llvm::APInt(I->Width, 1)};
+  }
+  InputSets.push_back(Cache);
+
+  for (auto &&I : Inputs) {
+    if (I->K == souper::Inst::Var)
+      Cache[I] = {llvm::APInt(I->Width, -1)};
   }
   InputSets.push_back(Cache);
 
+  for (auto &&I : Inputs) {
+    if (I->K == souper::Inst::Var)
+      Cache[I] = {llvm::APInt(I->Width, 0xFFF)};
+  }
+  InputSets.push_back(Cache);
+
+  std::srand(0);
+  for (int i = 0 ; i < 5; ++i ) {
+    for (auto &&I : Inputs) {
+      if (I->K == souper::Inst::Var)
+        Cache[I] = {llvm::APInt(I->Width, std::rand() % llvm::APInt(I->Width, -1).getLimitedValue())};
+    }
+    InputSets.push_back(Cache);
+  }
+
+  for (int i = 0 ; i < 5; ++i ) {
+    for (auto &&I : Inputs) {
+      if (I->K == souper::Inst::Var)
+        Cache[I] = {llvm::APInt(I->Width, std::rand() % 0xFFFFFFFF)};
+    }
+    InputSets.push_back(Cache);
+  }
```

Comment:
looks like you're not checking path conditions-- if you don't want to write that code, you need to leave a TODO or FIXME